Assets/MRTK/SDK/Editor/Inspectors/Utilities/Solvers/TapToPlaceInspector.cs

```diff
@@ -0,0 +1,125 @@
+        private SerializedProperty updateLinkedTransformProperty;
+        private SerializedProperty moveLerpTimeProperty;
+        private SerializedProperty rotateLerpTimeProperty;
+        private SerializedProperty scaleLerpTimeProperty;
+        private SerializedProperty maintainScaleProperty;
+        private SerializedProperty smoothingProperty;
+        private SerializedProperty lifetimeProperty;
+        private SerializedProperty magneticSurfaces;
+
+        private const string AdvancedPropertiesFoldoutKey = "TapToPlaceAdvancedProperties";
+
+        protected virtual void OnEnable()
+        {
+            instance = (TapToPlace)target;
+
+            // Main Tap to Place Properties
+            autoStart = serializedObject.FindProperty("autoStart");
+            defaultPlacementDistance = serializedObject.FindProperty("defaultPlacementDistance");
+            maxRaycastDistance = serializedObject.FindProperty("maxRaycastDistance");
+            useDefaultSurfaceNormalOffset = serializedObject.FindProperty("useDefaultSurfaceNormalOffset");
+            surfaceNormalOffset = serializedObject.FindProperty("surfaceNormalOffset");
+            keepOrientationVertical = serializedObject.FindProperty("keepOrientationVertical");
+            rotateAccordingToSurface = serializedObject.FindProperty("rotateAccordingToSurface");
+            debugEnabled = serializedObject.FindProperty("debugEnabled");
+            onPlacingStopped = serializedObject.FindProperty("onPlacingStopped");
+            onPlacingStarted = serializedObject.FindProperty("onPlacingStarted");
+
+            // Advanced Properties
+            updateLinkedTransformProperty = serializedObject.FindProperty("updateLinkedTransform");
+            moveLerpTimeProperty = serializedObject.FindProperty("moveLerpTime");
+            rotateLerpTimeProperty = serializedObject.FindProperty("rotateLerpTime");
+            scaleLerpTimeProperty = serializedObject.FindProperty("scaleLerpTime");
+            maintainScaleProperty = serializedObject.FindProperty("maintainScale");
+            smoothingProperty = serializedObject.FindProperty("smoothing");
+            lifetimeProperty = serializedObject.FindProperty("lifetime");
+            magneticSurfaces = serializedObject.FindProperty("magneticSurfaces");
+        }
+
+        public override void OnInspectorGUI()
+        {
+            RenderCustomInspector();
+        }
+
+        // Render the custom inspector with the basic and advanced properties
+        private void RenderCustomInspector()
+        {
+            serializedObject.Update();
+
+            EditorGUILayout.PropertyField(autoStart);
+            EditorGUILayout.PropertyField(defaultPlacementDistance);
+            EditorGUILayout.PropertyField(maxRaycastDistance);
+            EditorGUILayout.PropertyField(useDefaultSurfaceNormalOffset);
+
+            // Only show the SurfaceNormalOffset Property if UseDefaultSurfaceNormalOffset is false
```

Comment:
Nit: The comment itself is fairly obvious (i.e. UseDefaultSurfaceNormalOffset == false --> surfaceNormalOffset shows). This is an example of a case where a comment isn't adding a ton of value because the code is already readable (i.e. the english/naming is very obvious).

What is interesting to comment on here is why this is happening - i.e. SurfaceNormalOffset is only relevant when UseDefaultSurfaceNormalOffset is true (or you could negate that to make it match the code).

I want to encourage writing comments because they're super helpful, but want to highlight that the WHY is really important. The WHAT can still be important in more complicated cases (i.e. if you have a complex set of if/else conditions which you are maybe afraid to refactor because of legacy/low test coverage scenarios)

Reply:
Wow, I never really thought about code comments from this exact perspective, but I enthusiastically agree. Explaining _why_ something was done can be super helpful to an engineer down the road trying to determine if a change is safe. I like that.